### PR TITLE
[NeMo-UX] Add LLM and Lightning READMEs

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+NeMo LLM Collection
+===================
+
+The NeMo LLM Collection is the first collection that features NeMo 2.0, a redesign of NeMo that
```
Comment (jgerh) [inline nemo/collections/llm/README.md:4]: suggested revision.

The NeMo Framework LLM Collection introduces NeMo 2.0, a redesign that enhances the user experience by adopting a more PyTorch Lightning-like approach. This redesign aims to simplify NeMo and make it more modular.

File: nemo/collections/llm/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+aims to improve the user experience in by taking a more PyTorch Lightning-like approach to make
+NeMo both simpler and more modular.
```
Comment (jgerh) [inline nemo/collections/llm/README.md:6]: suggested revision.

The NeMo Framework LLM Collection introduces NeMo 2.0, a redesign that enhances the user experience by adopting a more PyTorch Lightning-like approach. This redesign aims to simplify NeMo and make it more modular.

File: nemo/collections/llm/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+NeMo LLM Collection
+===================
+
+The NeMo LLM Collection is the first collection that features NeMo 2.0, a redesign of NeMo that
+aims to improve the user experience in by taking a more PyTorch Lightning-like approach to make
+NeMo both simpler and more modular. 
+
+The current models are currently reimplemented in 2.0 as part of this collection:
```
Comment (jgerh) [inline nemo/collections/llm/README.md:8]: The following models are currently reimplemented in 2.0 as part of this collection:

File: nemo/collections/llm/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+- **GPT**
+- **LLaMA**
+- **Mixtral**
+
+For detailed tutorials and documentation of NeMo 2, please refer to the NeMo 2.0 [docs](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo_2.0/index.html) (TODO: update this hyperlink).
```
Comment (jgerh) [inline nemo/collections/llm/README.md:13]: Do not forget to remove TODO internal note. 

For detailed tutorials and documentation on NeMo 2.0, refer [docs](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo_2.0/index.html).

File: nemo/collections/llm/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+NeMo LLM Collection
```
Comment (jgerh) [inline nemo/collections/llm/README.md:1]: Please verify name. Which is correct: NeMo LLM Collection or NeMo Framework LLM Collection? Please make any changes as needed to the title.

File: nemo/lightning/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# NeMo Lightning
+
+This directory provides custom PyTorch Lightning-compatible objects for seamlessly training NeMo 2.0 models using PTL. NeMo 2.0 models
+are implemented using [Megatron Core](https://github.com/NVIDIA/Megatron-LM/tree/main/megatron/core), and NeMo Lightning provides the bridge between higher-level object-oriented PTL APIs and lower-level Megatron APIs. 
+For detailed tutorials and documentation of NeMo 2, please refer to the NeMo 2.0 [docs](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo_2.0/index.html) (TODO: update this hyperlink).
```
Comment (jgerh) [inline nemo/lightning/README.md:5]: Do not forget to remove TODO internal note.

For detailed tutorials and documentation on NeMo 2.0, refer to [docs](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo_2.0/index.html).

File: nemo/lightning/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+Some of the useful classes provided here are:
```
Comment (jgerh) [inline nemo/lightning/README.md:7]: Some of the helpful classes provided here include:

File: nemo/lightning/README.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+- [`MegatronStrategy`](./pytorch/strategies.py): A PTL strategy that enables training of Megatron models on NVIDIA GPUs. More information on `MegatronStrategy` can be found here (TODO: link to doc).
+- [`MegatronParallel`](./megatron_parallel.py): Class which sets up and manages Megatron's distributed model parallelism. Refer to this document for more information (TODO: Link to doc).
+- [`MegatronMixedPrecision`](./pytorch/plugins/mixed_precision.py): A specialized precision plugin for training Megatron-based models in PTL. More information can be found here (TODO: link to doc).
+- [`Trainer`](./pytorch/trainer.py): A lightweight wrapper around PTL's `Trainer` object which provides some additional support for capturing the arguments used to initialized the trainer. More information on NeMo 2's serialization mechanisms is available here (TODO: link to doc).
```
Comment (jgerh) [inline nemo/lightning/README.md:11]: Don't forget to remove TODO internal comment.

**Review by jgerh — COMMENTED** [review]
Completed technical edit of the files. Please review copyedits, make the changes in the source files, and resolve all threads.

File: nemo/collections/llm/README.md (hunk 2 above)
Comment (ashors1) [reply nemo/collections/llm/README.md]: I've verified that "NeMo LLM Collection" is correct.

**Review by jgerh — APPROVED** [review]
approved on request of @ashors1.

**Review by hemildesai — APPROVED** [review]

**Pull request merged**